endpoints/bookstore-grpc/bookstore.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+
+import status
+
+class Bookstore(object):
+  """Stores and manipulates bookstore data."""
+
+  class ShelfDict(dict):
```

Comment:
I played with this a bit, and think option 2 is the best which gives better errors with having to catch everywhere.

Let me know how you think.